### PR TITLE
Enable vibration calibration in other axis

### PR DESCRIPTION
**Pull request by theninj4**
Thanks for this awesome script and macro! I've been using it to fix some VFA on my v0.1 - I added another input parameter to let me specify different movement patterns for measuring either AB, XY, ABXY, A, B, X or Y.

You're welcome to take this or leave it, or it might be helpful to someone passing through.

--- 

Here are my results demonstrating the value. The A motion resonates at a different frequency to B, X and Y. Tomorrow I'll be checking all the bearings in that belt path 😉 

## A-axis
vibrations_20221005_234305

## B-axis
vibrations_20221005_235447

## X-Axis
vibrations_20221006_000230

## Y-Axis
vibrations_20221006_000937

## ABXY
vibrations_20221006_002545

### CODE REVIEW

File: macros/calibration/vibr_calibrate_02.cfg [hunk 1]
```diff
@@ -45,6 +45,7 @@ gcode:
     # PARAMETERS
     #
     {% set size = params.SIZE|default(60)|int %} # size of the area where the movements are done
+    {% set direction = params.DIRECTION|default(0)|int %} # 0 for XY, 1 for AB, 2 for ABXY, 3-6 for A/B/X/Y
```
Comment (theninj4) [inline macros/calibration/vibr_calibrate_02.cfg:48]: There muuuuuust be a better way of expressing this...

Comment (Frix-x) [reply macros/calibration/vibr_calibrate_02.cfg:48]: As stated in my comment below. I don't have a magic idea right now but using strings is probably a first step tower user-friendlyness of this macro.
I'll try to see if I have another idea :)

Comment (elpopo-eng) [reply macros/calibration/vibr_calibrate_02.cfg:48]: Hey, I'm thinking of something, give me some time to write this

Comment (elpopo-eng) [reply macros/calibration/vibr_calibrate_02.cfg:48]: My solution : 
```
[gcode_macro test_move]
description: Calibrate the printer flow or pressure advance
gcode:

    {% set size = params.SIZE|default(60)|int %} # size of the area where the movements are done
    {% set direction = params.DIRECTION|default('') %} # 0 for XY, 1 for AB, 2 for ABXY, 3-6 for A/B/X/Y
    {% set z_height = params.Z_HEIGHT|default(20)|int %} # z height to put the toolhead before starting the movements
    {% set verbose = params.VERBOSE|default(true) %} # Wether to log the current speed in the console
    
    {% set min_speed = params.MIN_SPEED|default(20)|int * 60 %} # minimum feedrate for the movements
    {% set max_speed = params.MAX_SPEED|default(200)|int * 60 %} # maximum feedrate for the movements
    {% set speed_increment = params.SPEED_INCREMENT|default(2)|int * 60 %} # feedrate increment between each move
    {% set feedrate_travel = params.TRAVEL_SPEED|default(200)|int * 60 %} # travel feedrate between moves

    {% set mid_x = printer.toolhead.axis_maximum.x|float / 2 %}
    {% set mid_y = printer.toolhead.axis_maximum.y|float / 2 %}



    {% set direction_factor = {
        'XY'  : { 
            'start' : {'x':  -0.5, 'y':  -0.5 },
            'move_factors' : {
                '0' : {'x':  0.5, 'y':  -0.5 },
                '1' : {'x':  0.5, 'y':   0.5 },
                '2' : {'x': -0.5, 'y':   0.5 },
                '3' : {'x': -0.5, 'y':  -0.5 }
                }
            },
        'AB' : { 
            'start' : {'x':  0.0, 'y':   0.0 },
            'move_factors' : {
                '0' : {'x':  0.5, 'y':  -0.5 },
                '1' : {'x': -0.5, 'y':   0.5 },
                '2' : {'x':  0.0, 'y':   0.0 },
                '3' : {'x':  0.5, 'y':   0.5 },
                '4' : {'x': -0.5, 'y':  -0.5 },
                '5' : {'x':  0.0, 'y':   0.0 }
                }
            },
        'ABXY' : { 
            'start' : {'x': -0.5, 'y':   0.5 },
            'move_factors' : {
                '0' : {'x': -0.5, 'y':  -0.5 },
                '1' : {'x':  0.5, 'y':  -0.5 },
                '2' : {'x': -0.5, 'y':   0.5 },
                '3' : {'x':  0.5, 'y':   0.5 },
                '4' : {'x': -0.5, 'y':  -0.5 },
                '5' : {'x': -0.5, 'y':   0.5 }
                }
            },
        'B'  : { 
            'start' : {'x':  0.5, 'y':   0.5 },
            'move_factors' : {
                '0' : {'x': -0.5, 'y':  -0.5 },
                '1' : {'x':  0.5, 'y':   0.5 }
                }
            },
        'A'  : { 
            'start' : {'x': -0.5, 'y':   0.5 },
            'move_factors' : {
                '0' : {'x':  0.5, 'y':  -0.5 },
                '1' : {'x': -0.5, 'y':   0.5 }
                }
            },
        'X'  : { 
            'start' : {'x': -0.5, 'y':   0.0 },
            'move_factors' : {
                '0' : {'x':  0.5, 'y':   0.0 },
                '1' : {'x': -0.5, 'y':   0.0 }
                }
            },
        'Y'  : { 
            'start' : {'x':  0.0, 'y':   0.5 },
            'move_factors' : {
                '0' : {'x':  0.0, 'y':  -0.5 },
                '1' : {'x':  0.0, 'y':   0.5 }
                }
            }            
        }
    %}

    {% if not (direction in direction_factor) %}
        { action_raise_error("Error check the directions") }
    {% endif %}

    # go to start position
    G90
    G1 Z{z_height}

    #RESPOND MSG="G1 X{mid_x + (size * direction_factor[direction].start.x) } Y{mid_y + (size * direction_factor[direction].start.y)} F{feedrate_travel}"
    G1 X{mid_x + (size * direction_factor[direction].start.x) } Y{mid_y + (size * direction_factor[direction].start.y)} F{feedrate_travel}


    {% for curr_speed in range(min_speed, max_speed + speed_increment) %}
        {% if (curr_speed - min_speed) % speed_increment == 0 %}
            {% if verbose %}
                RESPOND MSG="Current speed: {(curr_speed / 60)|int} mm/s"
            {% endif %}

            #ACCELEROMETER_MEASURE CHIP={accel_chip}
            {% for key, factor in direction_factor[direction].move_factors|dictsort %}
                #RESPOND MSG="factor = {factor}"
                #RESPOND MSG="G1 X{mid_x + (size * factor.x) } Y{mid_y + (size * factor.y)} F{curr_speed}"
                G1 X{mid_x + (size * factor.x) } Y{mid_y + (size * factor.y)} F{curr_speed}
            {% endfor %}
            #ACCELEROMETER_MEASURE CHIP={accel_chip} NAME=sp{(curr_speed / 60)|int}n1
            #G4 P300
            #M400
        {% endif %}
    {% endfor %}
```

Comment (Frix-x) [reply macros/calibration/vibr_calibrate_02.cfg:48]: This solution is exactly what I had in mind : something like like in [[gcode_macro _PROBE_MOVE_TO]](https://github.com/Frix-x/klipper-voron-V2/blob/5c8b46cae14b6c3e8a6b37195c0bd7c6f62e868d/macros/probing/dockable_probe.cfg#L56)

I think it's a good solution for now. Is this also OK for you @theninj4 ? We just need to add the Z axis in the dicts and we are good to go for this PR. Then I'll add the file naming if I find time to do it :)

**Comment by theninj4** [comment]
Neat 😄 Don't feel bound by the unwritten rules of open source to get my contributions in - as far as I'm concerned you've done all the heavy lifting here, if you'd like to copy + improve + commit on this you're most welcome to. Just wanted to show my appreciation of what you've got here so far and how it's helping me (and likely countless others!)

File: macros/calibration/vibr_calibrate_02.cfg (hunk 1 above)
Comment (theninj4) [reply macros/calibration/vibr_calibrate_02.cfg:48]: Looks good with me 👍

**Comment by Frix-x** [comment]
It's ok for me.
I like to keep track and credits, so if you want to do it, please do.
Else, I will do it by using and modifying directly files here in this PR to keep your contribution and the origin of the idea :)

**Comment by Frix-x** [comment]
This PR is replaced by #29

**Pull request closed without merging**